buildconfig/CMake/Eigen.cmake

```diff
@@ -15,7 +15,7 @@ else()
   # Download and unpack Eigen at configure time
   configure_file(${CMAKE_SOURCE_DIR}/buildconfig/CMake/Eigen.in ${CMAKE_BINARY_DIR}/extern-eigen/CMakeLists.txt)
 
-  execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" -DCMAKE_SYSTEM_VERSION=${CMAKE_SYSTEM_VERSION} . WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/extern-eigen )
+  execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" -DBUILD_TESTING="OFF" -DCMAKE_SYSTEM_VERSION=${CMAKE_SYSTEM_VERSION} . WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/extern-eigen )
```

Comment:
On my mac (no conda) I ran a configure in a clean build directory but I got a warning:
```
-- Using Eigen in ExternalProject
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
```
Do we need an updated version of eigen internally?

Reply:
We are currently using `3.3.4`, but whereas `3.3.9` is out now, so we are quite far behind. This might explain a recent problem I found where `liblapack` needed to be pinned to a previous version (had moved on in conda). I'll try moving it forward and see what breaks.